pkg/evaluator/builtin.go

```diff
@@ -597,6 +595,71 @@ func dashFunc(dashFn func([]float64)) BuiltinFunc {
 	}
 }
 
+var fontDecl = &parser.FuncDeclStmt{
+	Name:       "font",
+	Params:     []*parser.Var{{Name: "properties", T: parser.GENERIC_MAP}},
```

Comment:
is `GENERIC_MAP` correct here? I thought that was only for literals. For a formal argument list, I would have thought this should be `&Type{Name: MAP, Subtype: ANY_TYPE}`

Reply:
If I don't put this in I cant use:

```
font {size:10} 
```

as it is interpreted as `{}num`.
So yes, I believe this is as good as it gets for what I want here.

Reply:
I see. This seems to be a side-effect of how GENERIC works. so I guess this is fine for now, but I think the correct way is for `{size:10}` as a literal to remain GENERIC somehow so it can match either `{}num` or `{}any`. It is what we do for the empty composite literal, perhaps we need to do that for others?

Reply:
yes agreed!
but maybe not as part of this PR.